src/python/pants/backend/java/compile/javac.py

```diff
@@ -205,7 +205,7 @@ async def compile_java_source(
     # invoking via a `bash` wrapper (since the trailing portion of the command is executed by
     # the nailgun server). We might be able to resolve this in the future via a Javac wrapper shim.
     output_snapshot = await Get(Snapshot, Digest, compile_result.output_digest)
-    output_file = f"{request.component.representative.address.path_safe_spec}.jar"
+    output_file = f"{request.component.representative.address.path_safe_spec}.javac.jar"
```

Comment:
Maybe add a comment why the tool name is in the filename? (And same for the scalac filename.)

Reply:
Was on auto-merge. But yea, will do in a followup.